src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentStack.cs

```diff
@@ -316,6 +316,11 @@ public void PushRange(T[] items)
             {
                 throw new ArgumentNullException(nameof(items));
             }
+
+            // No op if the length is zero
+            if (items.Length == 0)
```

Comment:
If we're making this change I'd argue we should also do the same for `PushRange(array, 0, 0)` calls. That would also avoid the need for making this check twice.